lib/github-auth/acceptor.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+const request = require('request');
+const queryString = require('query-string');
+const serverSecrets = require('../server-secrets');
+const log = require('../log');
+
+function sendTokenToAllServers(token) {
+  const ips = serverSecrets.shieldsIps;
+  return Promise.all(ips.map(function(ip) {
+    return new Promise(function(resolve, reject) {
+      const options = {
+        url: 'https://' + ip + '/github-auth/add-token',
+        method: 'POST',
+        form: {
+          shieldsSecret: serverSecrets.shieldsSecret,
+          token: token,
+        },
+        // We target servers by IP, and we use HTTPS. Assuming that
+        // 1. Internet routers aren't hacked, and
+        // 2. We don't unknowingly lose our IP to someone else,
+        // we're not leaking people's and our information.
+        // (If we did, it would have no impact, as we only ask for a token,
+        // no GitHub scope. The malicious entity would only be able to use
+        // our rate limit pool.)
+        // FIXME: use letsencrypt.
+        strictSSL: false,
+      };
+      request(options, function(err, res, body) {
+        if (err != null) { return reject(err); }
+        resolve();
+      });
+    });
+  }));
+}
+
+function constEq(a, b) {
```

Comment:
I'm new to node in general but why not use [crypto.timingSafeEqual(a, b)](https://nodejs.org/api/crypto.html#crypto_crypto_timingsafeequal_a_b) instead?

Reply:
I didn't write this, just moved it. Yes, happy to switch that over. It seems like that solves the same problem.